### PR TITLE
fix for checkpoint saving issue in neurodamus-py:

### PR DESCRIPTION
**Pull request by WeinaJi**
NetCon with no target is allowed  in checkpoint
  set to pnttype = 0 and pntindex = -1

### CODE REVIEW

File: coreneuron/nrniv/nrn_checkpoint.cpp [hunk 1]
```diff
@@ -428,21 +428,26 @@ static void write_phase2(NrnThread& nt, FileHandlerWrap& fh) {
...
-        int ix = pnt->_i_instance;
-        if (ml_pinv[pnt->_type]) {
-            ix = ml_pinv[pnt->_type][ix];
-        }
+            // todo: this seems most natural, but does not work. Perhaps should look
+            // into how pntindex determined in nrnbbcore_write.cpp and change there.
+            int ix = pnt->_i_instance;
+            if (ml_pinv[pnt->_type]) {
+                ix = ml_pinv[pnt->_type][ix];
+            }
```
Comment (ohm314) [inline coreneuron/nrniv/nrn_checkpoint.cpp]: This code is in an `#if 0` should we not just drop it? If it's something that doesn't work, but still useful to keep maybe just make those 4 lines comments and remove the `#if 0`. @pramodk ?
In general we should rid the code of these ugly `#if 0` preproc branches.

**Comment by WeinaJi** [comment]
Simulation stack CI got stuck in build stage, after rerunning in https://bbpcode.epfl.ch/ci/job/hpc.SimulationStack/1158/ , everything passed.

**Comment by pramodk** [comment]
@WeinaJi @iomaganaris : will the simulation stack  failures will be fixed by #237?

**Comment by iomaganaris** [comment]
They will be fixed by my latest change in the `blueconfigs` branch in `gerrit`.

**Comment by iomaganaris** [comment]
please retest

**Pull request merged**